cluster-autoscaler/estimator/binpacking_estimator.go

```diff
@@ -68,49 +78,98 @@ func NewBinpackingNodeEstimator(
 // It is assumed that all pods from the given list can fit to nodeTemplate.
 // Returns the number of nodes needed to accommodate all pods from the list.
 func (e *BinpackingNodeEstimator) Estimate(
-	pods []*apiv1.Pod,
+	podsEquivalenceGroups []PodEquivalenceGroup,
 	nodeTemplate *schedulerframework.NodeInfo,
-	nodeGroup cloudprovider.NodeGroup) (int, []*apiv1.Pod) {
+	nodeGroup cloudprovider.NodeGroup,
+) (int, []*apiv1.Pod) {
 
-	e.limiter.StartEstimation(pods, nodeGroup, e.context)
+	e.limiter.StartEstimation(podsEquivalenceGroups, nodeGroup, e.context)
 	defer e.limiter.EndEstimation()
 
-	pods = e.podOrderer.Order(pods, nodeTemplate, nodeGroup)
-
-	newNodeNames := make(map[string]bool)
-	newNodesWithPods := make(map[string]bool)
+	podsEquivalenceGroups = e.podOrderer.Order(podsEquivalenceGroups, nodeTemplate, nodeGroup)
 
+	e.cleanHelperVariables()
 	e.clusterSnapshot.Fork()
 	defer func() {
 		e.clusterSnapshot.Revert()
 	}()
 
-	newNodeNameIndex := 0
 	scheduledPods := []*apiv1.Pod{}
-	lastNodeName := ""
+	for _, podsEquivalenceGroup := range podsEquivalenceGroups {
+		var err error
+		var remainingPods []*apiv1.Pod
+
+		scheduledPods, remainingPods, err = e.tryToScheduleOnExistingNodes(scheduledPods, podsEquivalenceGroup.Pods)
+		if err != nil {
+			klog.Errorf("%v", err)
```

Comment:
nit: This is equivalent to `klog.Errorf(err.Error())`, just with unnecessary formatting. Same below.